### PR TITLE
Add Identifiable protocol to the standard library

### PR DESCRIPTION
**Pull request by anandabits**
This PR implements SE-0261.  It will be updated with any changes after the core team makes a final decision.  Please let me know if it should target a different branch.

### CODE REVIEW

File: stdlib/public/core/Identifiable.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/// A class of types whose instances hold the value of an entity with stable identity.
+public protocol Identifiable {
```
Comment (airspeedswift) [inline stdlib/public/core/Identifiable.swift]: This (and the implementation) will need an `@available` annotation

File: stdlib/public/core/Identifiable.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  /// A type representing the stable identity of the entity associated with `self`.
+  associatedtype ID: Hashable
+
+  /// The stable identity of the entity associated with `self`.
+  var id: ID { get }
+}
+
+extension Identifiable where Self: AnyObject {
+  @inlinable
```
Comment (airspeedswift) [inline stdlib/public/core/Identifiable.swift]: Are we _certain_ this is definitively the implementation we'll want forever? If not, leaving off `@inlinable` may be a better choice, at least at first.

**Review by airspeedswift — CHANGES_REQUESTED** [review]

File: stdlib/public/core/Identifiable.swift [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/// A class of types whose instances hold the value of an entity with stable identity.
```
Comment (Azoy) [inline stdlib/public/core/Identifiable.swift]: This also needs the big copyright comment before anything else.

File: stdlib/public/core/Identifiable.swift (hunk 2 above)
Comment (anandabits) [reply stdlib/public/core/Identifiable.swift]: I'm fine with removing it for now.  I'm curious though - what other implementation options do you have in mind?

Comment (airspeedswift) [reply stdlib/public/core/Identifiable.swift]: ¯\_(ツ)_/¯

Mostly just making sure we've asked that question and are certain of the answer.

**Comment by anandabits** [comment]
@airspeedswift now that this has been accepted I’ll get the PR rebased on the latest master.  Are there any other changes that need to happen before it is ready to merge?  Additional documentation as mentioned in the review rationale?

**Comment by lorentey** [comment]
@anandabits Let's land the code as is -- we can figure out the docs later.

**Review by lorentey — APPROVED** [review]

**Comment by lorentey** [comment]
@swift-ci please smoke test

**Pull request merged**